src/main/java/com/hackclub/hccore/advancements/WolfAdv.java

```diff
@@ -0,0 +1,33 @@
+package com.hackclub.hccore.advancements;
+
+import com.fren_gor.ultimateAdvancementAPI.advancement.Advancement;
+import com.fren_gor.ultimateAdvancementAPI.advancement.BaseAdvancement;
+import com.fren_gor.ultimateAdvancementAPI.advancement.display.AdvancementDisplay;
+import com.fren_gor.ultimateAdvancementAPI.advancement.display.AdvancementDisplayBuilder;
+import com.fren_gor.ultimateAdvancementAPI.util.AdvancementKey;
+import com.fren_gor.ultimateAdvancementAPI.util.CoordAdapter;
+import com.hackclub.hccore.HCCorePlugin;
+import org.bukkit.Material;
+import org.bukkit.entity.EntityType;
+import org.bukkit.event.entity.EntityDeathEvent;
+
+public class WolfAdv extends BaseAdvancement {
+    static AdvancementDisplayBuilder<AdvancementDisplay.Builder, AdvancementDisplay> displayBuilder = new AdvancementDisplay.Builder(Material.BONE, "You Monster!")
+            .taskFrame()
+            .announceChat()
+            .showToast()
+            .description("Slaughter a doggo");
+    static int maxProgression = 1;
+
+    public WolfAdv(HCCorePlugin plugin, Advancement root, AdvancementKey key, CoordAdapter adapter) {
+        super(key.getKey(), displayBuilder.coords(adapter, key).build(), root, maxProgression);
+
+        registerEvent(EntityDeathEvent.class, e -> {
+            if (e.getEntityType() == EntityType.WOLF) {
+                if (e.getEntity().getKiller() != null) {
+                    incrementProgression(e.getEntity().getKiller());
```

Comment:
does this need checks to make sure its a player

Reply:
It shouldn't, as `.getKiller()` will either return the player killer or null.